src/main/grammars/RustParser.bnf

```diff
@@ -256,19 +256,15 @@ fake ValueParameterList ::= SelfParameter? ValueParameter* '...'? {
   elementTypeFactory = "org.rust.lang.core.stubs.StubImplementationsKt.factory"
 }
 
-FnParameters          ::= '(' !',' [ <<comma_separated_list FnParameter>> ] ')' { pin = 1 }
+FnParameters          ::= '(' !',' ( SelfParameter [ ',' [ <<comma_separated_list FnParameter>> ] ]
+                                   | [ <<comma_separated_list FnParameter>> ] ) ')' { pin = 1 }
```

Comment:
I think this two cases can be coalesced: ` [ SelfParameter (',' | &')') ] [ <<comma ... >>] ) `.